### PR TITLE
Vic 648 read consulting types from mongo db

### PR DESCRIPTION
**Pull request by tkuzynow**

### CODE REVIEW

File: src/main/java/de/caritas/cob/consultingtypeservice/ConsultingTypeServiceApplication.java [hunk 1]
```diff
@@ -7,6 +7,10 @@
 @SpringBootApplication(exclude = {DataSourceAutoConfiguration.class })
 public class ConsultingTypeServiceApplication {
 
+  static {
+    System.setProperty("os.arch", "i686_64");
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/ConsultingTypeServiceApplication.java]: What is the reason for setting this property here?

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java [hunk 2]
```diff
@@ -2,28 +2,33 @@
...
+  private @NonNull ConsultingTypeMongoRepository consultingTypeMongoRepository;
+  private @NonNull ConsultingTypeConverter consultingTypeConverter;
 
   /**
    * Get a complete list of all {@link ConsultingType}.
    * @return a {@link List} of {@link ConsultingType}
    */
   public List<ConsultingType> getListOfConsultingTypes() {
-    return new ArrayList<>(this.getConsultingTypesMap().values());
+    List<ConsultingTypeEntity> consultingTypeEntities = consultingTypeMongoRepository.findAll();
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: You could also use "var" for type

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java [hunk 3]
```diff
@@ -47,43 +56,34 @@ public ConsultingType getConsultingTypeById(Integer consultingTypeId) {
...
+  public void addConsultingType(ConsultingType consultingType) {
     if (isConsultingTypeWithGivenIdPresent(consultingType)
         || isConsultingTypeWithGivenSlugPresent(consultingType)) {
       LogService.logError(String
           .format("Could not initialize consulting type. id %s or slug %s is not unique",
               consultingType.getId(), consultingType.getSlug()));
       throw new UnexpectedErrorException();
     }
-    this.consultingTypesMap.put(consultingType.getId(), consultingType);
+    ConsultingTypeEntity consultingTypeEntity = new ConsultingTypeEntity();
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: You could also use "var" here

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoTenantAwareRepositoryService.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+  private @NonNull ConsultingTypeConverter consultingTypeConverter;
+
+  /**
+   * Get a complete list of all {@link ConsultingType}.
+   *
+   * @return a {@link List} of {@link ConsultingType}
+   */
+  public List<ConsultingType> getListOfConsultingTypes() {
+    if (isTechnicalTenantContext()) {
+      List<ConsultingTypeEntity> consultingTypeEntities = consultingTypeMongoTenantAwareRepository.findAll();
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoTenantAwareRepositoryService.java:38]: You could again use "var" in this class for types

**Review by maeventer — APPROVED** [review]

File: src/main/java/de/caritas/cob/consultingtypeservice/ConsultingTypeServiceApplication.java (hunk 1 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/ConsultingTypeServiceApplication.java]: accidentally commited, it's relevant for macOs (to be able to run some tests locally). I remove it, as it's not relevant for others

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoTenantAwareRepositoryService.java (hunk 4 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoTenantAwareRepositoryService.java:38]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java (hunk 3 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java (hunk 2 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: done.

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.stereotype.Repository;
+
+/**
+ * Repository for {@link ConsultingType} groups.
+ */
+@Repository
+@RequiredArgsConstructor
+public class ConsultingTypeGroupRepositoryImpl implements ConsultingTypeGroupRepository {
+
+  private @NonNull ConsultingTypeRepository consultingTypeRepository;
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: Injected beans should be declared as final

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  /**
+   * Get a map with all grouped consulting types.
+   *
+   * @return a {@link Map} with with the grouped {@link ConsultingType} instances.
+   */
+  @Cacheable(cacheNames = CacheManagerConfig.CONSULTING_TYPE_GROUPS_CACHE)
+  public Map<String, List<ConsultingType>> getConsultingTypesGroupMap() {
+    var consultingTypeEntities = consultingTypeRepository.findAll();
+    var consultingTypesGroupMap = new HashMap<String, List<ConsultingType>>();
+    consultingTypeEntities.stream().forEach(consultingTypeEntity ->
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: is it required to call the stream() function before using foreach()=

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      consultingType
+          .getGroups()
+          .forEach(group -> {
+            consultingTypesGroupMap.computeIfAbsent(group, newGroup -> new ArrayList<>());
+            consultingTypesGroupMap.get(group).add(consultingType);
+          });
+    }
+  }
+
+
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: Please remove unneccessary line breaks

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+@Repository
+@Primary
+@ConditionalOnExpression("${multitenancy.enabled:true}")
+@RequiredArgsConstructor
+@Slf4j
+public class ConsultingTypeGroupTenantAwareRepositoryImpl implements ConsultingTypeGroupRepository {
+
+  private static final Long TECHNICAL_TENANT_ID = 0L;
+  
+  private @NonNull ConsultingTypeTenantAwareRepository consultingTypeMongoTenantAwareRepository;
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java]: Please add also final declaration here

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+  /* it may not be cached, as the result is different for each tenant */
+  public Map<String, List<ConsultingType>> getConsultingTypesGroupMap() {
+    var consultingTypesGroupMap = new HashMap<String, List<ConsultingType>>();
+    findAllConsultingTypes().stream().forEach(consultingTypeEntity ->
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java]: Same question as before regarding stream().foreach()

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import org.springframework.stereotype.Repository;
+
+/**
+ * Repository for {@link ConsultingType}.
+ */
+@Repository("tenantUnaware")
+@RequiredArgsConstructor
+public class ConsultingTypeMongoRepositoryService implements ConsultingTypeRepositoryService {
+
+  private @NonNull ConsultingTypeRepository consultingTypeRepository;
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: Could you please also declare these two beans as final?

**Review by maeventer — APPROVED** [review]

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java (hunk 5 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java (hunk 9 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java]: done, skipped stream

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java (hunk 10 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeMongoRepositoryService.java]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java (hunk 8 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupTenantAwareRepositoryImpl.java]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java (hunk 7 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: done

File: src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java (hunk 6 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/consultingtypeservice/api/consultingtypes/ConsultingTypeGroupRepositoryImpl.java]: not required, skipping it. Thanks for the hint.

**Pull request merged**